keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,593 @@
+  
+- This KEP creates the new API version `local-registry.x-k8s.io/v1alpha1`,
+  This may open a can of worms. On the one hand, cluster implementations want
+  confidence that they're implementing a shared standard. On the other hand,
+  a new API version implies standards of governance, ownership, and API machinery
+  that might not make sense yet.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+None
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+Tools that configure a local registry should apply the following ConfigMap to
+the cluster:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-image-loaders
+  namespace: kube-public
+data:
+  registryStatus.v1: |
+    host: "localhost:5000"
+    hostFromCluster: "registry:5000"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+```
+
+Documentation that educates people on how to set up a registry manually should
+include instructions on how this is applied.
+
+The name of the config map must be `local-image-loaders`.
+
+The namespace must be `kube-public`.
+
+The `registryStatus.v1` field of the ConfigMap data must contain YAML. 
+The following code is the reference implementation that readers may use to parse the YAML:
+
+```go
+
+// A local registry allows clients to load images into the local cluster by
+// pushing to this registry.
+type LocalRegistryStatusV1 struct {
+	// Host documents the host (hostname and port) of the registry, as seen from
+	// outside the cluster.
+	//
+	// This is the registry host that tools should push images to.
+	Host string `yaml:"host,omitempty"`
+
+	// HostFromCluster documents the host (hostname and port) of the registry, as
+	// seen from the cluster's container runtime.
+	//
+	// When tools apply Kubernetes objects to the cluster, this host should be
+	// used for image name fields. If not set, users of this field should use the
+	// value of `host` instead.
+	HostFromCluster string `yaml:"hostFromCluster,omitempty"`
+
+	// Help contains a URL pointing to documentation for users that want to set up
+	// a registry.
+	//
+	// Tools can use this to nudge users to enable the registry. When possible,
+	// the writer should use as permanent a URL as possible to prevent drift
+	// (e.g., a version control SHA).
+	Help string `yaml:"help,omitempty"`
+}
+```
+
+The format of the data in `registryStatus.v1` should be considered frozen.
```

Comment:
i'd reword this to:
> The contents of a `registryStatus.vX` specification are frozen and will not change.

Reply:
the bellow details should be generalized and placed above the code examples / go struct.

the structure of the "design details" should be the following:
- general specification of where structures can be hosted - i.e. the ConfigMap name and namespace, without including details about `localRegistryStatus.v1`
- section for `localRegistryStatus.v1`
  - ConfigMap `data` example for `localRegistryStatus.v1`
  - Go struct specification for `LocalRegistryStatusV1`

Reply:
rearranged this a bit. Ya, I like this format better.